src/validate.ts

```diff
@@ -22,17 +22,19 @@ type customDatas = {[key: string]: any}
 
 interface QueryRequest{
     requestId: string;
-    inputs: Array<QueryRequest>;
+    inputs: {
+     intent: string; payload: { devices: { id: string; customData: customDatas; }[]; };
+    }[];
 }
 
-export function generateQueryRequest(deviceIds: string[], customDatas[]) : QueryRequest{
+export function generateQueryRequest(deviceIds: string[],customDatas: customDatas[]) : QueryRequest{
```

Comment:
Small nit, can we make custom interfaces and types TitleCase such as `CustomData` (omit the plural form since the array implies multiple). Typically custom types are TitleCase so that we can differentiate from built-in types and variable names.

Reply:
Done!